### ISSUE TITLE
How to use it as a custom loss function in tf?

### ISSUE THREAD

**Issue by wangshuo1994** [zzw922cn/TF2_soft_dtw#3]
Thank you very much for your share! Could you help me figure out how to implement the soft dtw as a custom loss in the deep learning training framework (in tf)? What I hope is to replace the RMSE/MAE.. metrics by the dtw. Thank you in advance.

**Comment by jackz314** [comment]
Did you ever figure out how to do this? I believe the goal is to wrap/transform the function into a subclass of [`Loss`](https://www.tensorflow.org/api_docs/python/tf/keras/losses/Loss), but I'm not familiar with TensorFlow or the implementation here enough to do this successfully yet.

**Comment by Christoper-Harvey** [comment]
@wangshuo1994 @jackz314 did either of you ever figure this out? I have been trying to implement this with TF2's custom gradient functions but haven't been able to get the backward pass to update anything. It is spitting out a value but the gradient is missing.

**Comment by JayKumarr** [comment]
https://github.com/AISViz/Soft-DTW
